ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -263,30 +250,144 @@ void PowerSystemNode::injectFault (const string& fault_name,
     {
       ros::param::getCached("/faults/" + FAULT_NAME_HPD, hpd_wattage);
       wattage += hpd_wattage;
-    } else
+    }
+  }
+}
+
+void PowerSystemNode::injectCustomFault(bool& fault_activated,
+                                        const PrognoserVector& sequence,
+                                        size_t& index,
+                                        double& wattage,
+                                        double& voltage,
+                                        double& temperature)
+{
+  static string saved_fault_directory = "N/A";
+  string current_fault_directory;
+  string designated_file;
+  static string saved_file;
+  static bool custom_fault_ready = false;
+  static bool custom_warning_displayed = false;
+  static bool end_fault_warning_displayed = false;
+  bool fault_enabled = false;
+
+  // Do nothing unless the specified fault has been injected.
+  if (! ros::param::getCached("/faults/activate_custom_fault", fault_enabled)) {
+    return;
+  }
+
+  if (!fault_activated && fault_enabled)
+  {
+    // Multiple potential points of failure, so alert user the process has started.
+    ROS_INFO_STREAM("Attempting custom fault activation...");
+
+    // Get user-entered file directory.
+    if (! ros::param::getCached("/faults/custom_fault_profile", designated_file))
+    {
+      ROS_ERROR("Failed to fetch custom fault profile name.");
+      return;
+    }
+
+    // Append the current fault directory to the stored file path.
+    current_fault_directory = ros::package::getPath("ow_power_system") + "/profiles/" + designated_file;
```

Comment:
This was a holdover from before when I was attempting to get directories full of fault profiles instead of just looking into a designated profiles directory. I'll change the variable name.